libyul/optimiser/UnusedAssignEliminator.cpp

```diff
@@ -74,84 +75,87 @@ void UnusedAssignEliminator::operator()(FunctionDefinition const& _functionDefin
 	UnusedStoreBase::operator()(_functionDefinition);
 }
 
+void UnusedAssignEliminator::operator()(FunctionCall const& _functionCall)
+{
+	UnusedStoreBase::operator()(_functionCall);
+
+	ControlFlowSideEffects sideEffects;
+	if (auto builtin = m_dialect.builtin(_functionCall.functionName.name))
```

Comment:
Hmm, unrelated, but this will need to be touched for `stdlib` since the builtins will be moved there, right? We'll likely need to expose the side effects of `stdlib` functions somehow if we plan on running the optimizer on them.

Reply:
These are assembly/yul built-ins. I think stdlib and the optimizer are orthogonal (but of course the usefulness of the stdlib depends on the efficiency of the optimizer).